### ISSUE TITLE
Specify CodeUri as a combination of bucket and key in SAM

### ISSUE THREAD

**Issue by Laxenade** [cloudtools/troposphere#848]
Hi, is it possible to use `Bucket` and `Key` in the `CodeUri` in SAM transformation. Currently, it only support passing `CodeUri` as a string.     
Example: [Serverless_S3_Processor.py](https://github.com/cloudtools/troposphere/blob/master/examples/Serverless_S3_Processor.py)
```json
CodeUri='s3://<bucket>/s3_processor.zip'
```
Thanks

**Comment by markpeek** [comment]
I'm not sure I understand the question. Are you asking if you can use an S3 URI such as s3://mybucket/folder/code.zip ? If so, the answer is yes.

**Comment by Laxenade** [comment]
My bad. Here is what I want to do (In SAM format):
```yaml
CodeUri:
         Bucket: <bucket>
         Key: <key>
```
And for now the template code only supports:
```yaml
CodeUri: <S3 Uri>
```

**Comment by phobologic** [comment]
https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#s3-location-object for more info.

**Comment by phobologic** [comment]
@Laxenade Troposphere as it stands only supports the basestring version - that said, it should be easy enough to modify the validator to take either.  If you want to submit a PR, we'd be happy to take a look!

**Comment by markpeek** [comment]
Fixed via #853
